File: PaymentSDKObjC/PaymentSDKObjC/AddCustomViewController.m
//
//  AddCustomViewController.m
//  PaymentSDKObjC
//
//  Created by Gustavo Sotelo on 11/05/16.
//  Copyright © 2016 Payment. All rights reserved.
//

#import "AddCustomViewController.h"
#import "UserModel.h"
#import <PaymentSDK/PaymentSDK-Swift.h>

@interface AddCustomViewController ()

@end

@implementation AddCustomViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self addPaymentWidgetToView:self. addView delegate:self uid:@"myuid" email:@"myemail"];
    
    
   
    self.PaymentAddVC = [PaymentSDKClient createAddWidget];
    [self addChildViewController:self.PaymentAddVC];
    self.PaymentAddVC.view.frame = CGRectMake(0, 0, self.view.bounds.size.width, self.addView.frame.size.height);
    self.addView.translatesAutoresizingMaskIntoConstraints = YES;
    [self.addView addSubview:self.PaymentAddVC.view];
    [self.PaymentAddVC didMoveToParentViewController:self];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)addCard:(UIButton*)sender {
    
    
    PaymentCard *validCard = [self.PaymentAddVC getValidCard];
    if (validCard != nil)
    {
        
        [self.activityIndicator startAnimating];
        [sender setEnabled:FALSE];
        
        [PaymentSDKClient add:validCard uid:USERMODEL_UID email:USERMODEL_EMAIL callback:^(PaymentSDKError *error, PaymentCard *cardAdded) {
            [sender setEnabled:YES];
            if(cardAdded != nil)
            {
                NSLog(@"%@",[cardAdded getJSONString]); // Print JSON sTRUCTURE
                dispatch_async(dispatch_get_main_queue(), ^{
                UIAlertController *alertC = [UIAlertController alertControllerWithTitle:@"Response" message:[NSString stringWithFormat:@"%@ Added, status:%@", cardAdded.termination, cardAdded.status] preferredStyle:UIAlertControllerStyleAlert];
                
                UIAlertAction *alertAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
                    [self.navigationController popViewControllerAnimated:FALSE];
                }];
                [alertC addAction:alertAction];
                [self presentViewController:alertC animated:NO completion:nil];
                });
            }
            else
            {
                dispatch_async(dispatch_get_main_queue(), ^{
                    UIAlertController *alertC = [UIAlertController alertControllerWithTitle:@"Error" message:[NSString stringWithFormat:@"code:%i, description:%@ help:%@",error.code, error.description, error.help]  preferredStyle:UIAlertControllerStyleAlert];
                    
                    UIAlertAction *alertAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
                        [self.navigationController popViewControllerAnimated:FALSE];
                    }];
                    [alertC addAction:alertAction];
                    [self presentViewController:alertC animated:NO completion:nil];
                });
            }
        }];
        
    }
    
}
- (IBAction)scanCard:(id)sender {
    
    [PaymentSDKClient scanCard:self callback:^(BOOL userClosed, NSString *cardNumber, NSString *expiryDate, NSString *cvv, PaymentCard *card) {
        
        if (!userClosed)
        {
            if (card != nil) // Handle card
            {
                dispatch_async(dispatch_get_main_queue(), ^{
                    UIAlertController *alertC = [UIAlertController alertControllerWithTitle:@"Response" message:[NSString stringWithFormat:@"%@ Card, cvv:%@, expiry:%@", card.termination, cvv, expiryDate] preferredStyle:UIAlertControllerStyleAlert];
                    
                    UIAlertAction *alertAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
                    }];
                    [alertC addAction:alertAction];
                    [self presentViewController:alertC animated:NO completion:nil];
                });
            }
                
        }
    }];
    
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: PaymentSDKSwift/PaymentSDKSwift/AddCustomViewController.swift
//
//  AddCustomViewController.swift
//  PaymentSDKSwift
//
//  Created by Gustavo Sotelo on 12/05/16.
//  Copyright © 2016 Payment. All rights reserved.
//

import UIKit
import PaymentSDK


class AddCustomViewController: UIViewController {
    
    @IBOutlet weak var addView: UIView!

    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
    
    var PaymentAddVC:PaymentAddNativeViewController!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        //Payment WIDGET CREATION
        
        PaymentAddVC = self.addPaymentWidget(toView: self.addView, delegate: nil, uid:UserModel.uid, email:UserModel.email)
        
        /* Change Colors
        PaymentAddVC.baseFontColor = .orange
        PaymentAddVC.baseColor = .green
        PaymentAddVC.backgroundColor = .white
        PaymentAddVC.showLogo = false*/
      PaymentAddVC.baseFont = UIFont(name: "Helvetica-Neue", size: 12) ?? UIFont.systemFont(ofSize: 12)
        
        // Do any additional setup after loading the view.
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
    
    @IBAction func addCard(_ sender:UIButton?)
    {
        if let validCard = self.PaymentAddVC.getValidCard() // CHECK IF THE CARD IS VALID, IF THERE IS A VALIDATION ERROR NIL VALUE WILL BE RETURNED
        {
            
            self.activityIndicator.startAnimating()
            sender?.isEnabled = false
            PaymentSDKClient.add(validCard, uid: UserModel.uid, email: UserModel.email, callback: { (error, cardAdded) in
                self.activityIndicator.stopAnimating()
                sender?.isEnabled = true
                if cardAdded != nil
                {
                    DispatchQueue.main.async(execute: {
                        let alertC = UIAlertController(title: "Response", message: "card "+cardAdded!.termination!+"  status:"+cardAdded!.status!, preferredStyle: UIAlertController.Style.alert)

                        let defaultAction = UIAlertAction(title: "Ok", style: UIAlertAction.Style.default, handler: { (action) in
                            
                            self.navigationController?.popViewController(animated: false)
                        })
                        alertC.addAction(defaultAction)
                        self.present(alertC, animated: true
                            , completion: {
                                
                        })
                    })
                }
                else
                {
                    DispatchQueue.main.async(execute: {
                        let alertC = UIAlertController(title: "Error", message: "error: "+error.debugDescription, preferredStyle: UIAlertController.Style.alert)
                        
                        let defaultAction = UIAlertAction(title: "OK", style: .default, handler: nil)
                        alertC.addAction(defaultAction)
                        self.present(alertC, animated: true
                            , completion: {
                                
                        })
                    })
                }
                
            })
        }
    }
}
